pkg/config/types/v1alpha1/v1alpha1_configurator.go

```diff
@@ -108,14 +108,12 @@ func (m *MachineConfig) Files() []machine.File {
 
 // Type implements the Configurator interface.
 func (m *MachineConfig) Type() machine.Type {
-	switch m.MachineType {
-	case "init":
-		return machine.TypeInit
-	case "controlplane":
-		return machine.TypeControlPlane
-	default:
+	t, err := machine.ParseType(m.MachineType)
+	if err != nil {
 		return machine.TypeWorker
 	}
+
+	return t
```

Comment:
Today we do no validation of `machine.type`, we just default to `TypeWorker` if `"init", or `"controlplane"` are not specified. That means that, today, users can throw just about anything they want in this field and it is still a worker. This fixes that.